analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -379,7 +393,9 @@ open class Npm(
 
         val id = Identifier("NPM", namespace, name, version)
 
-        if (packageDir.isSymbolicLink()) {
+        // While Yarn uses symbolic links only for workspace modules, PNPM uses symbolic links for all dependencies.
+        // Do not handle symbolic links differently when PNPM is using this method.
+        if (packageDir.isSymbolicLink() && managerName != "PNPM") {
```

Comment:
A potential way to handle this special treatment of PNPM could be to introduce a function `isWorkspaceDir(dir: File): Boolean` (or similar) in `Npm`. This could be overridden in `Pnpm`.

Reply:
Done. This also fixed a separate handling a few lines down:
```
val isWorkspaceSubmodule = packageFile.parentFile.realFile() in findWorkspaceSubmodules(workingDir)
if (hasIncompleteData && !isWorkspaceSubmodule) {
```